docs/tutorials/plugin.rst

```diff
@@ -0,0 +1,275 @@
+    ...
+
+Our plugin is disabled, for now. To enable it, we run::
+
+    $ tutor plugins enable myplugin
+    Plugin myplugin enabled
+    Configuration saved to /home/yourusername/.local/share/tutor/config.yml
+    You should now re-generate your environment with `tutor config save`.
+
+At this point you could re-generate your environment with ``tutor config save``, but there would not be any change to your environment... because the plugin does not do anything. So let's get started and make some changes.
+
+Modifying existing files with patches
+=====================================
+
+We'll start by modifying some of our Open edX settings files. It's a frequent requirement to modify the ``FEATURES`` setting from the LMS or the CMS in edx-platform. In the legacy native installation, this was done by modifying the ``lms.env.yml`` and ``cms.env.yml`` files. Here we'll modify the Python setting files that define the edx-platform configuration. To achieve that we'll make use of two concepts from the Tutor API: :ref:`patches` and :ref:`filters`.
+
+If you have not already read :ref:`how_does_tutor_work` now would be a good time :-) Tutor uses templates to generate various files, such as settings, Dockerfiles, etc. These templates include ``{{ patch("patch-name") }}`` statements that allow plugins to insert arbitrary content in there. These patches are located at strategic locations. See :ref:`patches` for more information.
+
+Let's say that we would like to limit access to our brandh new Open edX platform. It is not ready for prime-time yet, so we want to prevent users from registering new accounts. There is a feature flag for that in the LMS: `FEATURES['ALLOW_PUBLIC_ACCOUNT_CREATION'] <https://edx.readthedocs.io/projects/edx-platform-technical/en/latest/featuretoggles.html#featuretoggle-FEATURES['ALLOW_PUBLIC_ACCOUNT_CREATION']>`__. By default this flag is set to a true value, enabling anyone to create an account. In the following we'll set it to false.
+
+Add the following content to the ``myplugin.py`` file that you created earlier::
+
+    from tutor import hooks
+
+    hooks.filters.add_item(
+        hooks.Filters.ENV_PATCHES % "openedx-lms-common-settings",
+        "FEATURES['ALLOW_PUBLIC_ACCOUNT_CREATION'] = False"
+    )
+
+Let's go over these changes one by one::
+
+    from tutor import hooks
+
+This imports the ``hooks`` module from Tutor, which grants us access to ``hooks.filters`` and ``hooks.Filters`` (among other things).
+
+::
+
+    hooks.filters.add_item(
+        <filter>,
+        <item>
+    )
+
+This means "add <item> to the <filter>". In our case, we want to modify the LMS settings, both in production and development. The right patch for that is :patch:`openedx-lms-common-settings` and the corresponding filter is named ``hooks.Filters.ENV_PATCHES % "openedx-lms-common-settings"``. We add one item, which is a single Python-formatted line of code::
```

Comment:
Nit: I haven't seen the `%` operator in a while and actually forgot what this syntax does. I had to use the REPL to see what type of object `hooks.Filters.ENV_PATCHES` was before I understood this line 😅 

% strings aren't deprecated AFAIK but the python docs definitely encourage the newer `format` syntax. So I would suggest that something like this is a more modern approach:

```python
>>> hooks.Filters.env_patch = 'env:patches:{0}'.format
>>> hooks.Filters.env_patch("openedx-lms-common-settings")
'env:patches:openedx-lms-common-settings'
```
or even simply
```python
hooks.Filters.ENV_PATCH_PREFIX = "env:patches:"
hooks.Filters.ENV_PATCH_PREFIX + "openedx-lms-common-settings"
```
is more readable imho.

Reply:
Yes, you're right. I went with the `.format()` approach. But I also intend to add a "ENV_PATCH" filter that does not need to be string-formatted. Stay tuned!